src/Safebox.sol

```diff
@@ -0,0 +1,249 @@
+
+    /**
+     * @notice `usr` was granted owner access.
+     * @param usr The user address.
+     */
+    event Rely(address indexed usr);
+    /**
+     * @notice `usr` owner access was revoked.
+     * @param usr The user address.
+     */
+    event Deny(address indexed usr);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. The supported values are: "recipient".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The owner withdrawn assets from the safebox.
+     * @param recipient The token recipient.
+     * @param token The token withdrawn.
+     * @param amount The amount withdrawn.
+     */
+    event Withdraw(address indexed recipient, address indexed token, uint256 amount);
+    /**
+     * @notice A deposit was made into the safebox.
+     * @param sender The token sender.
+     * @param token The token deposited.
+     * @param amount The amount deposited.
+     */
+    event Deposit(address indexed sender, address indexed token, uint256 amount);
+    /**
+     * @notice `usr` was granted custodian access.
+     * @param usr The user address.
+     */
+    event AddCustodian(address indexed usr);
+    /**
+     * @notice `usr` custodian access was revoked.
+     * @param usr The user address.
+     */
+    event RemoveCustodian(address indexed usr);
+    /**
+     * @notice The recipient has been sucessfully changed.
+     * @param recipient The new recipient address.
+     */
+    event RecipientChange(address indexed recipient);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "Safebox/not-ward");
+        _;
+    }
+
+    modifier onlyCustodian() {
+        require(custodians[msg.sender] == 1, "Safebox/not-custodian");
+        _;
+    }
+
+    /**
+     * @param _vat The MCD vat module.
+     * @param _owner The safebox owner.
+     * @param _custodian The safebox custodian.
+     * @param _recipient The recipient for assets in the safebox.
+     */
+    constructor(address _vat, address _owner, address _custodian, address _recipient) {
+        require(_recipient != address(0), "Safebox/invalid-recipient");
+
+        wards[_owner] = 1;
+        emit Rely(_owner);
+
+        custodians[_custodian] = 1;
+        emit AddCustodian(_custodian);
+
+        vat = VatLike(_vat);
+        recipient = _recipient;
+    }
+
+    /*//////////////////////////////////
+                 Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Withdraws ERC20-like tokens from this contract.
+     * @dev Anyone can call this function after MakerDAO governance executes an Emergency Shutdown.
+     * @param token The token to be withdrawn.
+     * @param amount The amount of tokens.
+     */
+    function withdraw(address token, uint256 amount) external {
+        require(wards[msg.sender] == 1 || vat.live() == 0, "Safebox/not-ward");
```

Comment:
I'm tempted to refactor this into:

```suggestion
        if (vat.live() == 1) {
            require(wards[msg.sender] == 1, "Safebox/not-ward");
        }
```

This is less gas efficient, but clearer on the intent.

Reply:
If `vat.live()` is 0 it'll skip this check. If you want to refactor this just make sure that all cases are still checked.

Reply:
That is the expected behavior.

This is an escape hatch in case of Emergency Shutdown: we will allow anyone to withdraw the funds from this contract into the `recipient`.